app/plugin_api.go

```diff
@@ -285,6 +285,10 @@ func (api *PluginAPI) SearchChannels(teamId string, term string) (*model.Channel
 	return api.app.SearchChannels(teamId, term)
 }
 
+func (api *PluginAPI) SearchUsers(search *model.UserSearch, options *model.UserSearchOptions) ([]*model.User, *model.AppError) {
```

Comment:
This API is now a bit strange because we are passing two structs that appear to have the same purpose. I looked into it a bit more and UserSearchOptions is more of an internal thing that should probably be combined with UserSearch. 

For the plugin API I think we should only accept `model.UserSearch` and here we can derive the UserSearchOptions. We can just always search as an administrator for now. You can see how the options map here: https://github.com/mattermost/mattermost-server/blob/master/api4/user.go#L554

Reply:
Ok let me update that.

Reply:
Just to be sure both `plugin/api.go` and `app/plugin_api.go` both will have one parameter `model.UserSearch`?

Reply:
Right. But you have to do do the translation in `plugin_api.go` because `api.app.SearchUsers` takes both parameters. (See my note above)

Reply:
Ok, got it updated the PR

Reply:
@crspeller ^